modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -241,30 +215,46 @@ private void onWriteBusy(Iterator<CommandClosure<WriteCommand>> iterator) {
+                boolean[] applied = {false};
+
                 if (command instanceof UpdateCommand) {
-                    result = handleUpdateCommand((UpdateCommand) command, commandIndex, commandTerm);
+                    result = handleUpdateCommand((UpdateCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof UpdateAllCommand) {
-                    result = handleUpdateAllCommand((UpdateAllCommand) command, commandIndex, commandTerm);
+                    result = handleUpdateAllCommand((UpdateAllCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof FinishTxCommand) {
-                    result = handleFinishTxCommand((FinishTxCommand) command, commandIndex, commandTerm);
+                    result = handleFinishTxCommand((FinishTxCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof WriteIntentSwitchCommand) {
-                    handleWriteIntentSwitchCommand((WriteIntentSwitchCommand) command, commandIndex, commandTerm);
+                    handleWriteIntentSwitchCommand((WriteIntentSwitchCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof SafeTimeSyncCommand) {
-                    handleSafeTimeSyncCommand((SafeTimeSyncCommand) command, commandIndex, commandTerm);
+                    handleSafeTimeSyncCommand((SafeTimeSyncCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof BuildIndexCommand) {
-                    handleBuildIndexCommand((BuildIndexCommand) command, commandIndex, commandTerm);
+                    handleBuildIndexCommand((BuildIndexCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof PrimaryReplicaChangeCommand) {
-                    handlePrimaryReplicaChangeCommand((PrimaryReplicaChangeCommand) command, commandIndex, commandTerm);
+                    handlePrimaryReplicaChangeCommand((PrimaryReplicaChangeCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof VacuumTxStatesCommand) {
-                    handleVacuumTxStatesCommand((VacuumTxStatesCommand) command, commandIndex, commandTerm);
+                    handleVacuumTxStatesCommand((VacuumTxStatesCommand) command, commandIndex, commandTerm, applied);
                 } else if (command instanceof UpdateMinimumActiveTxBeginTimeCommand) {
-                    handleUpdateMinimalActiveTxTimeCommand((UpdateMinimumActiveTxBeginTimeCommand) command, commandIndex, commandTerm);
+                    handleUpdateMinimalActiveTxTimeCommand((UpdateMinimumActiveTxBeginTimeCommand) command, commandIndex, commandTerm,
+                            applied);
                 } else {
                     assert false : "Command was not found [cmd=" + command + ']';
                 }
+
+                if (applied[0]) {
```

Comment:
Agree with Roma, that decreases the readability.

Reply:
How do you suggest to fix this? Java doesn't support returning tuples unfortunately.

Reply:
You may introduce a special class to return.

Reply:
Refactored to return IgniteBiTuple.
Not sure if it became better.

Reply:
Well, IgniteBiTuple doesn't sound as a special class, why not to introduce result wrapper something like
```
CommandProcessingResult
  Serializable result
  Boolean applied
```
explaining in javadoc why the class is needed and what applied means?

Reply:
The tuple works fine to me.

Reply:
> IgniteBiTuple<Serializable, Boolean> result = null;

It's quit difficult to understand what that means. At least it deserves a comment.

Reply:
Ok, I'll add a comment.